### DISCUSSION TITLE
Version field not created/or incremented.

### DISCUSSION THREAD

**Discussion by Dragomir-Ivanov** [mikro-orm/mikro-orm#5778] [General]
### Describe the bug

Greetings,
I tried very hard to find the solution myself, but still to no avail.

I have the following BaseEntity: 

```
@Entity({ abstract: true })
export abstract class BaseEntity {
  @Property()
  @TransformDate()
  createdAt: Date = new Date();

  @Property({
    onUpdate: () => new Date(),
  })
  @TransformDate()
  updatedAt: Date = new Date();

  @Property({ nullable: true, version: true })
  version: number | null;
}
```

Basically I have MirkoORM setup with NestJS, and PostgreSQL DB driver. Using auto generated PostgreSQL repository.

Everything else seems to work, except that `version` is not created, nor incremented automatically when I first initialize it to 1. 
Also `updatedAt` is not updated on each update, but stays the initial value, same as `createdAt`.

Here is my `Service` code:

```
    const entity = await this.customerRepository.findOneOrFail({ id });
    
    /// Make entity changes here with regular asignments
    /// entity.fieldname = "updated";

    return await this.customerRepository.upsert(entity);
```

### Reproduction

See above.

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.5

### Node.js version

v20.13.1

### Operating system

Linux

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
What's the point of upserting a managed entity? Do you understand this won't do anything? You need to flush for this to have any effect, upsert is not meant to be used this way, if you pass a managed entity to it, it won't do anything else than `em.assign()` on it, and here you don't even provide any data. You should just `em.flush()`.

I will check how optimistic locking is handled for the `em.upsert` call, but what you showed in the code shouldn't fire any query. Also you say the version field is not even created - but your entity already exists, it should have it set already (and the `em.flush` call should increment it if it sees any changes to be flushed).

**Reply by B4nan** [reply]
This is the code path that should be triggered if you pass a managed entity to it:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/EntityManager.ts#L875

**Reply by Dragomir-Ivanov** [reply]
Thank you @B4nan. 
I changed the code below and it really did work for both field, `updatedAt` and `version`.

```
    const entity = await this.customerRepository.findOneOrFail({ id });
    
    /// Make entity changes here with regular asignments
    /// entity.fieldname = "updated";

    this.customerRepository.getEntityManager().flush();
    return entity;
```

I am sorry for bothering you. I am new to MikroORM and whole NestJS world. 
I will investigate a bit `upsert` path as well. 
When doing plain `em.flush()`, I guess I need to find the entity again, to obtain it's auto-generated properties.

**Reply by B4nan** [reply]
`em.flush` will flush the in progress changes of all managed entities, so you first need to have some in the context - either new ones that you mark with `em.persist` (or you create them via `em.create` which calls `em.persist` automatically), or loaded ones.

`em.upsert` is handy when you don't know if something exists - so it makes no sense to use that after you load an entity from database, especially with `em.findOneOrFail` since that means your program will crash if the entity is non existent.

**Reply by Dragomir-Ivanov** [reply]
Thank you for the valuable insights @B4nan !

I will ask some more things on Discord, not to waste your time further.
